docs: Support building documentation locally

bdrung opened the pull request:
Building the documentation locally without Internet access will fail, because sphinx will try to download the intersphinx inventory from http://docs.python.org/3/objects.inv.

Support building the documentation locally without Internet access by searching for locally installed intersphinx inventory first.

Bug-Debian: https://bugs.debian.org/755026

waynew reviewed (changes requested):
At least on Arch, there's a different path that should be added:

```
diff --git a/doc/conf.py b/doc/conf.py
index 4b96bae628..40154b21ee 100644
--- a/doc/conf.py
+++ b/doc/conf.py
@@ -40,6 +40,7 @@ intersphinx_mapping = {
             "/usr/share/doc/python{}.{}/html/objects.inv".format(
                 sys.version_info[0], sys.version_info[1]
             ),
+            "/usr/share/doc/python/html/objects.inv",
             None,
         ),
     )
```

That allowed me to build while offline

bdrung committed:
docs: Support building documentation locally

Building the documentation locally without Internet access will fail,
because sphinx will try to download the intersphinx inventory from
http://docs.python.org/3/objects.inv.

Support building the documentation locally without Internet access by
searching for locally installed intersphinx inventory first.

Bug-Debian: https://bugs.debian.org/755026
Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung commented:
Okay. I included `/usr/share/doc/python/html/objects.inv` in the merge request.

bdrung commented:
Just a rebase on master (to hopefully get a green CI).
